other/autoregressive-process/utils.jl

```diff
@@ -0,0 +1,42 @@
+# Generates an AR(p) process with coefficients `ϕ`.
+# `ϕ` should be provided as a vector and it represents the coefficients of the AR model.
+# Hence the order of the generated process is equal to the length of `ϕ`.
+# `s` indicates the total length of the series to be generated.
+function generate_process(ϕ::AbstractVector{Float32}, s::Int)
+    @assert s > 0 "s must be positive"
+    # Generate white noise
+    ϵ = randn(Float32, s)
+    # Initialize time series
+    X = zeros(Float32, s)
+    p = length(ϕ)
+    X[1] = ϵ[1]
+    # Reverse the order of the coefficients for multiplication later on
+    reverse!(ϕ) 
+    # Compute values iteratively
+    for t ∈ 2:s
+        if t ≤ p
+            X[t] = sum(X[1:t-1] .* ϕ[1:t-1]) + ϵ[t]
+        else
+            X[t] = sum(X[t-p:t-1] .* ϕ) + ϵ[t]
+        end
+    end
+    X
+end
+
+# Create batches of a time series `X` by splitting the series into
+# sequences of length `s`. Each new sequence is shifted by `r` steps.
+# When s == r,  the series is split into non-overlapping batches.
+function batch_timeseries(X, s::Int, r::Int)
```

Comment:
Is there no existing functionality for this in https://github.com/JuliaML/MLUtils.jl? If not, perhaps there should be.

Reply:
I believe there is not, this is also something I want to add to https://fluxml.ai/Flux.jl/stable/models/recurrence/#Working-with-sequences but I was unsure on whether it was wanted/useful. I also have a similar function to reverse the batched time series into a tabular format. If there is already a functionality, I'm happy to learn about it.